src/renderer/dx/DxRenderer.cpp

```diff
@@ -997,6 +1000,49 @@ try
 
         _d2dRenderTarget->BeginDraw();
         _isPainting = true;
+
+        {
+            // Get the baseline for this font as that's where we draw from
+            DWRITE_LINE_SPACING spacing;
+            RETURN_IF_FAILED(_dwriteTextFormat->GetLineSpacing(&spacing.method, &spacing.height, &spacing.baseline));
+
+            // GH#5098: If we're rendering with cleartype text, we need to always
+            // render onto an opaque background. If our background's opacity is
+            // 1.0f, that's great, we can use that. Otherwise, we need to force the
+            // text renderer to render this text in grayscale. In
+            // UpdateDrawingBrushes, we'll set the backgroundColor's a channel to
+            // 1.0 if we're in cleartype mode and the background's opacity is 1.0.
+            // Otherwise, at this point, the _backgroundColor's alpha is <1.0.
+            //
+            // Currently, only text with the default background color uses an alpha
+            // of 0, every other background uses 1.0
+            //
+            // DANGER: Layers slow us down. Only do this in the specific case where
+            // someone has chosen the slower ClearType antialiasing (versus the faster
+            // grayscale antialiasing)
+            const bool usingCleartype = _antialiasingMode == D2D1_TEXT_ANTIALIAS_MODE_CLEARTYPE;
+            const bool usingTransparency = _defaultTextBackgroundOpacity != 1.0f;
+            // Another way of naming "bgIsDefault" is "bgHasTransparency"
+            const auto bgIsDefault = (_backgroundColor.a == _defaultBackgroundColor.a) &&

```

Comment:
I'm pretty sure we can't just set this one once at the start of the frame - I _think_ we'll need to compute it whenever the BG color changes.

Reply:
Alright, I'll go through this on Wednesday.

Reply:
OK, I moved this to a helper and I make it update the drawing context when the color changes. That should resolve it. If I end up making the runs hold multiple colors and condensing them, I'll have to revisit again to store this information too. But until then, I think it should be fine. I'm testing it now.